tools/requests_gun/internal/trigger/trigger.go

```diff
@@ -0,0 +1,100 @@
+package trigger
+
+import (
+	"log"
+	"os"
+	"os/signal"
+	"requests_gun/internal/ammo_provider"
+	"requests_gun/internal/parse_args"
+	"requests_gun/internal/request_maker"
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+func Fire(ammoProvider *ammo_provider.AmmoProvider, args *parse_args.CliArgs) {
+	interrupt := make(chan os.Signal, 1)
+	signal.Notify(interrupt, os.Interrupt)
+
+	wg := sync.WaitGroup{}
+	ticker := time.NewTicker(time.Second)
+	for {
+		select {
+		case s := <-interrupt:
+			log.Println("Got signal:", s)
+			log.Println("Stopping...")
+			ticker.Stop()
+			return
+		case <-ticker.C:
+			statistics := statistics{startTime: time.Now(), printErrors: args.PrintErrors}
+			do_shot := func() {
+				defer wg.Done()
+				bullet := ammoProvider.GetBullet()
+				requestMaker := request_maker.NewHttp(args.Url, args.Port)
+				response_data, err := requestMaker.MakeRequest(bullet)
+				statistics.add(response_data, err)
+			}
+
+			second_start := time.Now()
+			requests_number := uint(0)
+			for requests_number < args.TargetLoad && time.Since(second_start) < time.Second {
+				wg.Add(1)
+				go do_shot()
+				requests_number++
+			}
+			wg.Wait()
+			if time.Since(second_start) > time.Second {
+				log.Println("Requests took longer than a second, probably need to decrease the load.")
+			}
+			statistics.print()
+		}
+	}
+}
+
+type counters struct {
+	totalRequestsCount atomic.Uint64
```

Comment:
Also a minor thing, but it's clear from the struct name that the attributes are counters. So the `Count` suffix is redundant.